controller/volume_controller.go

```diff
@@ -972,15 +970,22 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, es map[stri
 
 		if v.Status.ExpansionRequired {
 			if v.Spec.Size == e.Status.CurrentSize {
-				if !v.Spec.DisableFrontend && v.Spec.Frontend == types.VolumeFrontendBlockDev {
-					// Best effort. We don't know if there is a file system built in the volume.
-					if err := util.ExpandFileSystem(v.Name); err != nil {
-						logrus.Warnf("failed to expand the file system for the volume %v: %v", v.Name, err)
-					} else {
-						logrus.Infof("Succeeded to expand the file system for the volume %v", v.Name)
+				// Start the blockdev frontend and expand the filesystem.
+				if v.Spec.Frontend == types.VolumeFrontendBlockDev {
+					v.Status.FrontendDisabled = false
+					e.Spec.DisableFrontend = false
+					if e.Status.Endpoint != "" {
```

Comment:
Add a comment on the implicit requeue here, like https://github.com/longhorn/longhorn-manager/blob/f5faefaf09fbc25ea3d99cd001c197199be1f488/controller/volume_controller.go#L961

After spec.DisableFrontend changed, the engine will be updated before reaching `Endpoint != ""`